src/Servers/Kestrel/Core/src/Internal/Infrastructure/BodyControl.cs

```diff
@@ -70,10 +71,17 @@ public Task StopAsync()
             return _responseWriter.StopAcceptingWritesAsync();
         }
 
-        public void Abort(Exception error)
+        public void Abort(IOException error)
         {
-            _requestReader.Abort(error);
-            _emptyRequestReader.Abort(error);
+            _requestReader.Abort(new IOException(error.Message, error));
+            _emptyRequestReader.Abort(new IOException(error.Message, error));
+            _responseWriter.Abort();
+        }
+
+        public void Abort(ConnectionAbortedException error)
+        {
+            _requestReader.Abort(new ConnectionAbortedException(error.Message, error, error.CancellationToken));
+            _emptyRequestReader.Abort(new ConnectionAbortedException(error.Message, error, error.CancellationToken));
```

Comment:
I want to take this change just without the changes to new up two read Exceptions in all the abort paths. I think it's unnecessary, since unlike with writing, there can really only one reader at a time to observer the exception.

Reply:
Depends if Edi.Capture can be called on both paths?

Reply:
Depends if Edi.Capture can be called on both paths; both capturing into the same exception?

Reply:
That said, maybe it's not best to use this change as a starting point, since it's main purpose was copying the Exceptions. I'll open another PR that just changes Http2StreamErrorException to derive from IOException, since I think that's an easy win.

Reply:
Probably should do this one though https://github.com/aspnet/AspNetCore/pull/11875 as it currently will create junky exceptions if it throws more than once after abort

Reply:
I just noticed that it was the *InnerException* that wasn't an IOException which I'm completely OK with. Please ignore this thread 😆